### PR TITLE
Integrate secrets and registries migration

### PR DESCRIPTION
**Pull request by tboerger**

### CODE REVIEW

File: migrate/secrets.go [hunk 1]
```diff
@@ -2,10 +2,96 @@ package migrate
...
+		log := logrus.WithFields(logrus.Fields{
+			"repo":   secretV0.RepoFullname,
+			"secret": secretV0.Name,
+		})
+
+		log.Debugln("migrate secret")
+
+		repoV1 := &RepoV1{}
+
+		if err := meddler.QueryRow(target, repoV1, fmt.Sprintf(repoSlugQuery, secretV0.RepoFullname)); err != nil {
```
Comment (bradrydzewski) [inline migrate/secrets.go:40]: I believe you can search by `sercretV0.RepoID` since we preserve the ID

File: migrate/secrets.go [hunk 2]
```diff
@@ -2,10 +2,96 @@ package migrate
...
+		log.Debugln("migration complete")
+	}
+
+	logrus.Infof("migration complete")
+	return tx.Commit()
 }
+
+const secretImportQuery = `
+SELECT
+	repo_full_name,
```
Comment (bradrydzewski) [inline migrate/secrets.go:81]: I think you can remove this field / the inner join and instead you can lookup the repository using `secret.secret_repo_id`

File: migrate/repos.go [hunk 3]
```diff
@@ -93,48 +91,44 @@ func MigrateRepos(source, target *sql.DB) error {
...
+
 		remoteRepo, _, err := client.Repositories.Find(ctx, scm.Join(repo.Namespace, repo.Name))
+
 		if err != nil {
 			log.WithError(err).Errorf("failed to get remote repository")
 			multierror.Append(result, err)
 			continue
 		}
 
-		// 2.c update the temporary id for the remote
```
Comment (bradrydzewski) [inline migrate/repos.go:135]: I think having some of these comments in place are important.  It is probably non-obvious what we are doing here without an explanation

File: migrate/secrets.go (hunk 1 above)
Comment (tboerger) [reply migrate/secrets.go:40]: At least for my MariaDB test databases I had the case if mismatch with the ids...

File: migrate/secrets.go (hunk 2 above)
Comment (tboerger) [reply migrate/secrets.go:81]: I need to try it again, but as I mentioned at https://github.com/drone/drone-migrate/pull/2#discussion_r250267760 I had mismatches for the ids, that's why I started to use the fullnames.

**Comment by bradrydzewski** [comment]
thanks!  I had some minor comments, but there was no reason to block the PR for them, so you can just ignore :)

**Pull request merged**